src/Endatix.Core/Entities/Submission.cs

```diff
@@ -29,15 +28,11 @@ public Submission(string jsonData, long? formDefinitionId = null, bool isComplet
             FormDefinitionId = formDefinitionId.Value;
         }
 
-        IsComplete = isComplete;
         JsonData = jsonData;
         CurrentPage = currentPage;
         Metadata = metadata;
 
-        if (isComplete)
-        {
-            CreatedAt = DateTime.UtcNow;
-        }
+        SetCompletionData(isComplete);
```

Comment:
Just a suggestion, but I think `SetCompletionStatus` will be more natural. Every method updates data, so data is redundant term. Status refers to the business meaning of the Completion related data attributes. What do you think?

Reply:
Makes sense, thanks.